mobile/ios/ios-core.ts

```diff
@@ -407,16 +424,39 @@ export class CoreFoundation implements  Mobile.ICoreFoundation {
 		return this.dictionaryCreate(null, keysBuffer, valuesBuffer, len, this.kCFTypeDictionaryKeyCallBacks(), this.kCFTypeDictionaryValueCallBacks());
 	}
 
-	private cfTypeTo(value: NodeBuffer): NodeBuffer {
-		var typeId = this.getTypeID(value);
-		var retval: NodeBuffer = null;
+	public cfTypeTo(dataRef: NodeBuffer): any {
+		var typeId = this.getTypeID(dataRef);
 
-		if(typeId === this.dataGetTypeID()) {
-			var len = this.dataGetLength(value);
-			retval = ref.reinterpret(this.dataGetBytePtr(value), len);
-		}
+		if(typeId === this.stringGetTypeID()) {
+			return this.convertCFStringToCString(dataRef);
+		} else if(typeId === this.dataGetTypeID()) {
+			var len = this.dataGetLength(dataRef);
+			var retval = ref.reinterpret(this.dataGetBytePtr(dataRef), len);
+			return retval;
+		} else if(typeId === this.dictionaryGetTypeID()) {
+			var count = this.dictionaryGetCount(dataRef);
 
-		return retval;
+			var keys = new Buffer(count * CoreTypes.pointerSize);
+			var values = new Buffer(count * CoreTypes.pointerSize);
+			this.dictionaryGetKeysAndValues(dataRef, keys, values);
+
+			var jsDictionary = Object.create(null);
+			var offset = 0;
+
+			for(var i=0; i<count; i++) {
+				var keyPointer = ref.readPointer(keys, offset, CoreTypes.pointerSize);
+				var valuePointer = ref.readPointer(values, offset, CoreTypes.pointerSize);
+				offset += CoreTypes.pointerSize;
+
+				var jsKey = this.cfTypeTo(keyPointer);
+				var jsValue = this.cfTypeTo(valuePointer);
+				jsDictionary[jsKey] = jsValue;
+			}
+
+			return jsDictionary;
+		} else { // We don't need it for now
+			return "";
```

Comment:
Currently we are reading all properties for the application, but actually we use only two of them - `Path` and `CFBundleExecutable` (both of them are strings).  We don't need the another properties so we don't need to convert them. If we throw error here we won't be able to execute `device list-applications` command  because it will throw error every time. So I think we shouldn't throw error here.